### PR TITLE
[+] mongo service name resolver for multi cluster mongo 

### PR DESCRIPTION
**Pull request by skcc321**
the provided patch gives opportunity to define different senvice_name for multicluster mongo based on standard definition:
```ruby
    c.use :mongo, describes: /localhost:27017/ do |mongo|
      mongo.service_name = "mongo_sts"
    end

    c.use :mongo, describes: /localhost:27018/ do |mongo|
      mongo.service_name = "mongo_iot"
    end
```

### CODE REVIEW

File: lib/ddtrace/contrib/mongodb/instrumentation.rb
Comment (marcotc) [inline lib/ddtrace/contrib/mongodb/instrumentation.rb:54]: It seems like [`Mongo::Address#to_param` is not a valid method](https://app.circleci.com/pipelines/github/DataDog/dd-trace-rb/3044/workflows/e2e5682d-8d15-4224-b956-ec7b98666972/jobs/121786#failed-test-476).
We could probably use `Mongo::Address#seed` here to get the same kind of output.

**Review by marcotc — COMMENTED** [review]
Thank you so much for you contribution, @skcc321!
You've found the correct places to make the change you want, I left an inline comment about one of the lines you changed.

Also, it would good to have tests asserting that it is now possible to use the `describes` configuration with the MongoDB integration. Given we only have a single MongoDB instance (host:port) for testing, I suggest using two clients, `localhost:27017` and `127.0.0.1:27017`, in order to perform distinct requests that can be separated into different configuration sets.

File: lib/ddtrace/contrib/mongodb/instrumentation.rb
Comment (skcc321) [reply lib/ddtrace/contrib/mongodb/instrumentation.rb]: Yes, you are right... somehow it worked on my local with ruby 2.7 - I changed it to seed.
Anyway, I'll try to add specs for two mongo clients as you suggested soon.
Right now, unfortunately, I cannot get the CI test run green even with existing specs...

Comment (ivoanjo) [reply lib/ddtrace/contrib/mongodb/instrumentation.rb]: Apologies for the CI brokenness @skcc321. I've raised this internally, we'll look into this ASAP as it's breaking for every PR. I'll leave a quick note here once it's fixed.

Comment (ivoanjo) [reply lib/ddtrace/contrib/mongodb/instrumentation.rb]: 😓  a few more things broke due to upstream releases, we're still working to get CI back to green.

Comment (ivoanjo) [reply lib/ddtrace/contrib/mongodb/instrumentation.rb]: CI is now back to green 🎉  . A rebase on top of current `master` should get you a nice starting state :)

Comment (skcc321) [reply lib/ddtrace/contrib/mongodb/instrumentation.rb]: still some failures. But anyway, testing is possible now

**Comment by skcc321** [comment]
@marcotc added related specs.

**Comment by marcotc** [comment]
I think the implementation is all good, thank you!

One final thing before we can merge it is documenting it.
Could you add a few examples to the MongoDB section here? https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#mongodb
One example for simple string matching, e.g. `describes: 'host:port'`, and another for a regular expression matching, e.g. `describes: /host.*/`, would suffice.

Take a look at the Redis and Net::HTTP sections of the documentation for examples.

**Comment by skcc321** [comment]
Got. Will be ready soon

**Comment by skcc321** [comment]
@marcotc done.

**Comment by skcc321** [comment]
Screenshot 2021-04-08 at 00 15 42

**Review by marcotc — APPROVED** [review]
Awesome work, @skcc321! Thank you!

I'll make sure to rebase the PR when we address the build failures and will merge it for you after that.

**Comment by skcc321** [comment]
So nothing :)

**Comment by marcotc** [comment]
I'm so sorry, @skcc321.
I'll merge your work for the next release.

**Comment by skcc321** [comment]
Thank you @marcotc. I'll keep an eye on it 😉

**Review by marcotc — APPROVED** [review]
Thank you so much for your work, @skcc321! 🙇

**Comment by marcotc** [comment]
👋 Thank you again @skcc321, https://github.com/DataDog/dd-trace-rb/releases/tag/v0.54.0 was just released, including this change.

**Pull request merged**
